### PR TITLE
Adding 2 additional tests to subscription for selectable customizable…

### PR DESCRIPTION
**Pull request by latran**
This is for my feature area for the selectable customizable column.  Included is 2 test to check the functionality of the filtering column checkboxes and verifying the column headers.

### CODE REVIEW

File: tests/foreman/ui_airgun/test_subscription.py [hunk 1]
```diff
@@ -310,3 +310,48 @@ def test_positive_view_vdc_guest_subscription_products(session):
             content_products = session.subscription.content_products(
                 VDC_SUBSCRIPTION_NAME, virt_who=True)
             assert content_products and product_name in content_products
+
+@tier2
+def test_select_customizable_columns_uncheck_all_checkboxes(session):
+    """ Ensure that user can uncheck all checkboxes
+
```
Comment (vijay8451) [inline tests/foreman/ui_airgun/test_subscription.py]: Test case ID missing:
i.e. https://github.com/SatelliteQE/robottelo/blob/8a03aa7f4d6714a312993356253c2d232211e1a0/tests/foreman/ui_airgun/test_subscription.py#L61

File: tests/foreman/ui_airgun/test_subscription.py [hunk 2]
```diff
@@ -310,3 +310,48 @@ def test_positive_view_vdc_guest_subscription_products(session):
...
+    """
+    with session:
+        session.subscription.reset_scc()
+        col_head_count = session.subscription.read_header()
+        assert len(col_head_count) == 1 and col_head_count == ('Select all rows',)
+
+@tier2
+def test_select_customizable_columns_check_all_checkboxes(session):
+    """ Ensure that user can uncheck all checkboxes
+
```
Comment (vijay8451) [inline tests/foreman/ui_airgun/test_subscription.py]: same as above.

**Comment by latran** [comment]
Added changed to file.  Redid implementation.

**Comment by vijay8451** [comment]
> Added changed to file. Redid implementation.

Thanks, Could you share the test result too.

**Comment by latran** [comment]
Test 1:

'(pyenv) [ltran@ltran-rdu-fedora robottelo]$ pytest tests/foreman/ui_airgun/test_subscription.py::test_select_customizable_columns_uncheck_all_checkboxes
================ test session starts ================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.7.0, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ltran/Projects/robottelo, inifile:
plugins: xdist-1.24.1, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.0
collecting ... 2019-02-18 10:39:18 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                                                  

tests/foreman/ui_airgun/test_subscription.py .                                                                                                                                                              [100%]

============= 1 passed in 21.61 seconds ============='

Test 2:
`(pyenv) [ltran@ltran-rdu-fedora robottelo]$ pytest tests/foreman/ui_airgun/test_subscription.py::test_select_customizable_columns_check_all_checkboxes
================ test session starts ================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.7.0, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ltran/Projects/robottelo, inifile:
plugins: xdist-1.24.1, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.0
collecting ... 2019-02-18 10:37:21 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                                                  

tests/foreman/ui_airgun/test_subscription.py .                                                                                                                                                              [100%]

============= 1 passed in 22.55 seconds =============`

**Comment by oshtaier** [comment]
Do you have request to implement that functionality or maybe client found defect associated with such actions? Because, in general, these test cases provide small coverage, out of scope for functional testing, out of scope for robotello initial idea, completely out of scope for new airgun UI approach and definitely are not tier2 tests.

**Comment by latran** [comment]
@oshtaier The functionality of this feature area is that a user can filter the Subscriptions table header columns with the 'selectable customizable column' checkboxes.  I can expand more coverage of this functionality by adding filtering of one column at a time and checking the subscription table header again, but if that's worth the effort.

Another functionality is checking the available data entry instead of the header columns after filtering or being unable to click into the details page when filtering out the 'Name' column. Would that be better in terms of functionality/test coverage?

Any advice is helpful since this feature area is just about filtering the Subscriptions Table

**Comment by oshtaier** [comment]
okay, if filtering is actually the only functionality that you need then we have to follow your way. It is just not the best candidate for automation

**Comment by omkarkhatavkar** [comment]
@latran I agree on the comment of @oshtaier  https://github.com/SatelliteQE/robottelo/pull/6718#issuecomment-469170347, this is not the best candidate for automation. If this really required, I would add both check and uncheck filter in a single test to reduce the time of browser openning and navigation. 

@latran  What do you say?

**Comment by oshtaier** [comment]
yeah, do not merge till airgun code resolved. There are many concerns here

**Comment by JacobCallahan** [comment]
Since this feature is entirely in the UI, having some automation around it is preferred to none at all.
With that, I do agree that this can be compressed down to one test that covers all the checks in these two.

It could also be nice to have a test that sets a specific selection of columns, then either refreshes or navigates away and back to the page. The test verification should be that the columns match the specified state.

**Comment by latran** [comment]
Latest test results from run.

(pyenv) [ltran@ltran-rdu-fedora robottelo]$ pytest tests/foreman/ui_airgun/test_subscription.py::test_select_customizable_columns_uncheck_and_checks_all_checkboxes
========================================== test session starts 
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.7.0, pluggy-0.8.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ltran/Projects/robottelo, inifile:
plugins: xdist-1.24.1, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.0
collecting ... 2019-04-29 10:11:50 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                                                                                                       

tests/foreman/ui_airgun/test_subscription.py .                                                                                                                                                                                                                   [100%]

======================================= 1 passed in 39.33 seconds

**Comment by latran** [comment]
Can I get another review for this?

**Pull request merged**